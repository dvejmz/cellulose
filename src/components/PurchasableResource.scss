.resource-purchasable {
    .resource {
        display: inline-block;
    }
}
