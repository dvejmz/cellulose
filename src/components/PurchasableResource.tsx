import React from 'react';
import Resource, { ResourceProps } from './Resource';

import './PurchasableResource.scss';

type PurchasableResourceProps = ResourceProps &
  {
    onBuyClick: () => void;
    price: number;
    currency: string;
  };

const PurchasableResource: React.FC<PurchasableResourceProps> = (props: PurchasableResourceProps) => {
  return (
    <div className={`resource-purchasable`} data-test-id={`resource-purchasable-${props.id}`}>
      <Resource {...props} />
      <button
        className={`resource-purchasable__buy-button btn btn-sm`}
        onClick={props.onBuyClick}
      >
        Buy <span className="resource-purchasable__buy-price">({`${props.currency}${props.price}`})</span>
      </button>
    </div>
  );
};

export default PurchasableResource;
