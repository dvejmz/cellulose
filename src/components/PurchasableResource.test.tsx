import { shallow } from 'enzyme';
import React from 'react';
import PurchasableResource from './PurchasableResource';

describe('<PurchasableResource />', () => {
  let purchasableResource: any;

  beforeEach(() => {
    purchasableResource = shallow(
      <PurchasableResource 
        name="Pulp"
        id="pulp"
        quantity={9001}
        quantityUnit="kg"
        purchaseRate={1}
        currency="£"
        price={10}
        onBuyClick={() => {}} // tslint:disable-line no-empty
      />);
  });

  it('renders ok', () => (
    expect(purchasableResource).toMatchSnapshot()
  ));

  it('shows buy text', () => (
    expect(purchasableResource.find('.resource-purchasable__buy-button').text()).toContain('Buy')
  ));

  it('shows resource price', () => (
    expect(purchasableResource.find('.resource-purchasable__buy-price').text()).toContain('(£10)')
  ));
});
