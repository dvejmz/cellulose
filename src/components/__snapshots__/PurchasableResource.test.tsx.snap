// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PurchasableResource /> renders ok 1`] = `ShallowWrapper {}`;
